samples/subsys/video/capture/src/main.c

```diff
@@ -57,43 +53,72 @@ int main(void)
 		const struct video_format_cap *fcap = &caps.format_caps[i];
 		/* fourcc to string */
 		printk("  %c%c%c%c width [%u; %u; %u] height [%u; %u; %u]\n",
-		       (char)fcap->pixelformat,
-		       (char)(fcap->pixelformat >> 8),
-		       (char)(fcap->pixelformat >> 16),
-		       (char)(fcap->pixelformat >> 24),
-		       fcap->width_min, fcap->width_max, fcap->width_step,
-		       fcap->height_min, fcap->height_max, fcap->height_step);
+		       (char)fcap->pixelformat, (char)(fcap->pixelformat >> 8),
+		       (char)(fcap->pixelformat >> 16), (char)(fcap->pixelformat >> 24),
+		       fcap->width_min, fcap->width_max, fcap->width_step, fcap->height_min,
+		       fcap->height_max, fcap->height_step);
 		i++;
 	}
 
 	/* Get default/native format */
-	if (video_get_format(video, VIDEO_EP_OUT, &fmt)) {
+	if (video_get_format(video_dev, VIDEO_EP_OUT, &fmt)) {
 		LOG_ERR("Unable to retrieve video format");
 		return 0;
 	}
 
 	printk("- Default format: %c%c%c%c %ux%u\n", (char)fmt.pixelformat,
-	       (char)(fmt.pixelformat >> 8),
-	       (char)(fmt.pixelformat >> 16),
-	       (char)(fmt.pixelformat >> 24),
-	       fmt.width, fmt.height);
+	       (char)(fmt.pixelformat >> 8), (char)(fmt.pixelformat >> 16),
+	       (char)(fmt.pixelformat >> 24), fmt.width, fmt.height);
+
+#if DT_HAS_CHOSEN(zephyr_display)
+	struct display_capabilities capabilities;
+	struct display_buffer_descriptor buf_desc;
+	const struct device *display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device %s not found", display_dev->name);
+		return 0;
+	}
+
+	printk("\nDisplay device: %s\n", display_dev->name);
+
+	display_get_capabilities(display_dev, &capabilities);
+
+	printk("- Capabilities:\n");
+	printk("  x_resolution = %u, y_resolution = %u, supported_pixel_formats = %u\n"
+	       "  current_pixel_format = %u, current_orientation = %u\n\n",
+	       capabilities.x_resolution, capabilities.y_resolution,
+	       capabilities.supported_pixel_formats, capabilities.current_pixel_format,
+	       capabilities.current_orientation);
+
+#if defined(CONFIG_VIDEO_MCUX_MIPI_CSI2RX)
```

Comment:
I think we should try to avoid interspersing device specific Kconfigs within the sample as much as possible- Perhaps we could define a few constants at the top of this file based on which video devices were enabled, like so?

```
#ifdef CONFIG_VIDEO_MCUX_MIPI_CSI2RX
#define DISPLAY_FORMAT PIXEL_FORMAT_ARGB_8888
#else
#define DISPLAY_FORMAT PIXEL_FORMAT_RGB565
#endif

#ifdef CONFIG_MCUX_ELCID_PXP
#define BUFF_ALIGN 64
#else
#define BUFF_ALIGN 1
#endif
```

My reasoning here is that this allows us to keep all the driver-specific KConfigs in one location within the sample, so that snippets like these don't get interspersed within the sample to handle specific hardware requirements

Reply:
Yes, done. Just a small remark, when there is no alignment required, it is actually `sizeof(void *)` not 1 :-).